controller/utils.go

```diff
@@ -0,0 +1,46 @@
+package controller
+
+import (
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+func hasReplicaEvictionRequested(rs map[string]*longhorn.Replica) bool {
```

Comment:
Suggest moving to replica_controller.go. 

It's better to have a good convention to place this util function, especially they are resource related. For example, you can add them next to resource controller, or even create another pkg folder to aggregate different types of resource util functions.

I would not suggest having a misc uitl.go to place these functions, because it will cause some confusion and is not a good convention to arrange code placement.

Reply:
I see your point for not preferring having a mixed until.go file. However, these non-method function are being used by multiple controller so would it make sense to have them together inside one file `util.go` ? I am not very strong about this though. I am ok with either approach (keeping them in a mixed util.go file OR moving them back to related controller). WDYT @innobead @shuo-wu ?

Reply:
Being used in different controllers should be fine, but what I expect is the way to place codes should be easy to understand and remember. For example, this function name is related to replica, so it can be simply placed in the replica controller and this will be a convention connection.  

As I mentioned [here](https://github.com/longhorn/longhorn-manager/pull/1541#discussion_r1193378314), we go with the current code if all of you agree on.